daml-lf/spec/daml-lf-1.rst

```diff
@@ -2218,179 +2377,177 @@ as described by the ledger model::
+     'fetch' @Mod:T cid ‖ (st; keys)
        ⇓ᵤ
-     Ok (vₜ, ε) ‖ E ; (st; keys)
+     Ok (vₜ, ε) ‖ (st; keys)
 
-      e ‖ E₀  ⇓  Ok vₖ ‖ E₁
+      e  ⇓  Ok vₖ
       (Mod:T, vₖ) ∈ dom(keys₀)      cid = keys((Mod:T, v))
       st(cid) = (Mod:T, vₜ, 'active')
    —————————————————————————————————————————————————————————————————————— EvUpdFetchByKeyFound
-     'fetch_by_key' @Mod:T e ‖ E₀ ; (st; keys)
+     'fetch_by_key' @Mod:T e ‖ (st; keys)
         ⇓ᵤ
-     Ok ⟨'contractId': cid, 'contract': vₜ⟩ ‖ E₁ ; (st; keys)
+     Ok ⟨'contractId': cid, 'contract': vₜ⟩ ‖ (st; keys)
 
      'tpl' (x : T) ↦ { …, 'key' @σ eₖ eₘ }  ∈  〚Ξ〛Mod
-     e ‖ E₀  ⇓  Ok vₖ ‖ E₁
-     (eₘ vₖ) ‖ E₁  ⇓  vₘ ‖ E₂
+     e  ⇓  Ok vₖ
+     (eₘ vₖ)  ⇓  vₘ
      (Mod:T, vₖ) ∉ dom(keys₀)
     —————————————————————————————————————————————————————————————————————— EvUpdFetchByKeyNotFound
-     'fetch_by_key' @Mod:T e ‖ E₀ ; (st; keys)
+     'fetch_by_key' @Mod:T e ‖ (st; keys)
         ⇓ᵤ
-     Err "Lookup key not found"  ‖ E₂ ; (st; keys)
+     Err "Lookup key not found"  ‖ (st; keys)
 
      'tpl' (x : T) ↦ { …, 'key' @σ eₖ eₘ }  ∈  〚Ξ〛Mod
-     e ‖ E₀  ⇓  Ok vₖ ‖ E₁
-     (eₘ vₖ) ‖ E₁  ⇓  vₘ ‖ E₂
+     e  ⇓  Ok vₖ
+     (eₘ vₖ)  ⇓  vₘ
      (Mod:T, vₖ) ∈ dom(keys)   cid = keys((Mod:T, v))
    —————————————————————————————————————————————————————————————————————— EvUpdLookupByKeyFound
-     'look_by_key' @Mod:T e ‖ E₀ ; (st; keys)
+     'look_by_key' @Mod:T e ‖ (st; keys)
        ⇓ᵤ
-     Ok ('Some' @(Contract:Id Mod:T) cid) ‖ E₁ ; (st; keys)
+     Ok ('Some' @(Contract:Id Mod:T) cid) ‖ (st; keys)
 
      'tpl' (x : T) ↦ { …, 'key' @σ eₖ eₘ }  ∈  〚Ξ〛Mod
-     e ‖ E₀  ⇓  Ok vₖ ‖ E₁
-     (eₘ vₖ) ‖ E₁  ⇓  vₘ ‖ E₂
+     e  ⇓  Ok vₖ
+     (eₘ vₖ)  ⇓  vₘ
      (Mod:T, vₖ) ∉ dom(keys)
    —————————————————————————————————————————————————————————————————————— EvUpdLookupByKeyNotFound
-     'look_by_key' @Mod:T e ‖ E₀ ; (st; keys)
+     'look_by_key' @Mod:T e ‖ (st; keys)
          ⇓ᵤ
-     Ok ('None' @(Contract:Id Mod:T)) ‖ E₁ ; (st; keys)
+     Ok ('None' @(Contract:Id Mod:T)) ‖ (st; keys)
 
      LitTimestamp is the current ledger time
    —————————————————————————————————————————————————————————————————————— EvUpdGetTime
-     'get_time' ‖ E ; (st; keys)
+     'get_time' ‖ (st; keys)
        ⇓ᵤ
-     Ok (LitTimestamp, ε) ‖ E ; (st; keys)
+     Ok (LitTimestamp, ε) ‖ (st; keys)
 
-     e  ‖ E₀  ⇓  Ok u ‖ E₁
-     u ‖ E₁ ; st₀  ⇓ᵤ  ur ‖ E₂ ; st₁
+     e   ⇓  Ok u
+     u ‖ st₀  ⇓ᵤ  ur ‖ st₁
    —————————————————————————————————————————————————————————————————————— EvUpdEmbedExpr
-     'embed_expr' @τ e ‖ E₀; st₀  ⇓ᵤ  ur ‖ E₂ ; st₁
+     'embed_expr' @τ e; st₀  ⇓ᵤ  ur ‖ st₁
 
 
-Similar to expression evaluation, we do not explicitly specify the
-cases where sub-expressions fail. Those case can be inferred in a
-straightforward way by following the left-to-right evaluation order.
+We do not explicitly specify the cases where sub-expressions fail.
```

Comment:
What happens in these cases?

Reply:
I will specify this -- was waiting for some feedback first. :-)

Reply:
Yeah, I only read the PR description after reviewing the code. It's not the first time that happens to me. I should change my habits in this regard... Apologies.